common/persistence/sql/sqlplugin/mysql/plugin.go

```diff
@@ -97,11 +111,67 @@ func (p *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
 	if cfg.MaxConnLifetime > 0 {
 		db.SetConnMaxLifetime(cfg.MaxConnLifetime)
 	}
+
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
 	return db, nil
 }
 
+func registerTLSConfig(cfg *config.SQL) error {
+	host, _, err := net.SplitHostPort(cfg.ConnectAddr)
+	if err != nil {
+		return fmt.Errorf("error in host port from ConnectAddr: %v", err)
+	}
+
+	tlsConfig := &tls.Config{
+		ServerName: host, // hostname
+	}
+
+	if cfg.TLS.CaFile != "" {
+		rootCertPool := x509.NewCertPool()
+		pem, err := ioutil.ReadFile(cfg.TLS.CaFile)
+		if err != nil {
+			return fmt.Errorf("failed to load CA files: %v", err)
+		}
+		if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+			return fmt.Errorf("failed to append CA file")
+		}
+		tlsConfig.RootCAs = rootCertPool
+	}
+
+	if cfg.TLS.CertFile != "" && cfg.TLS.KeyFile != "" {
+		clientCert := make([]tls.Certificate, 0, 1)
+		certs, err := tls.LoadX509KeyPair(
+			cfg.TLS.CertFile,
+			cfg.TLS.KeyFile,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to load key pair: %v", err)
+		}
+		clientCert = append(clientCert, certs)
+		tlsConfig.Certificates = clientCert
+	}
+
+	// Register a customer tls configuration
+	err = mysql.RegisterTLSConfig(customTLSName, tlsConfig)
+	if err != nil {
+		return fmt.Errorf("failed to register tls config: %v", err)
+	}
+
+	if cfg.ConnectAttributes == nil {
+		cfg.ConnectAttributes = map[string]string{}
+	}
+
+	if cfg.ConnectAttributes["tls"] == customTLSName || cfg.ConnectAttributes["tls"] == "" {
```

Comment:
I'm confused by this check.  Maybe you intended the following:
```
if cfg.ConnectAttributes["tls"] != customTLSName || cfg.ConnectAttributes["tls"] == "" {
```
or just maybe:
```
if cfg.ConnectAttributes["tls"] == "" {
```

Reply:
Maybe it will help if you can include an example of config when someone specifies tls as ConnectAttributes.

Reply:
I removed the error and used the simpler check you suggested. 

The thinking was that in order to correctly use a tls config you need to register it and tell the connection to use it via a connect attribute. I'm automatically setting the connect attribute here so you don't need to also provide that when using a tls connection. The error originally was intended for a case where someone provided a _different_ tls connect attribute for some reason and also provided a tls config. In that case the tls config will silently not get used. Not sure if that really constitutes an error here.